python/GafferUI/ColorChooser.py

```diff
@@ -106,15 +108,148 @@ def _displayTransformChanged( self ) :
 		GafferUI.Slider._displayTransformChanged( self )
 		self._qtWidget().update()
 
+class _ColorFieldWidget( QtWidgets.QWidget ) :
+
+	def __init__( self, parent = None ) :
+
+		QtWidgets.QWidget.__init__( self, parent )
+		self.__size = 194 # The height of the sliders and swatches in a `ColorChooser` dialogue.
+
+	def resizeEvent( self, event ) :
+
+		w = event.size().width()
+		h = event.size().height()
+
+		if w != h :
+			self.__size = h
+			self.setMinimumWidth( h )
+
+	def sizeHint( self ) :
+
+		# This is needed to set our size for layout when a dialog using `ColorChooser` is first
+		# opened. Without it, the other widgets in our row would be shrunk considerably.
+
+		return QtCore.QSize( self.__size, self.__size )
```

Comment:
How useful do you think the expanding-in-y size behaviour is in practice? When resizing a colour dialogue, it leaves a lot of dead space above and below the sliders (we had the same problem before, but I'd like to fix it), and leads to some juddery behaviour at times. It also seems likely to be responsible for the resizing judder that has been introduced when popping up an editor in the LightEditor, and the subsequent non-centering of the popup on the cursor. Perhaps we're better off not expanding in Y at all?

Reply:
I think it might be fairly useful, but for now I've made it a fixed size. Since we're just testing and reviewing the functionality, I'll leave it to when we do the full color chooser layout to sort out the sizing behavior.